src/indexer/indexer.service.ts

```diff
@@ -20,98 +20,110 @@ export type TransactionOutput = {
+        if (0x52 <= firstOpCode && firstOpCode <= 0x60) return undefined;
+    }
 
-        // extract the input public keys from the transaction
-        const pubKeys: Buffer[] = [];
-        for (const input of vin) {
-            const pubKey = extractPubKeyFromScript(
-                Buffer.from(input.prevOutScript, 'hex'),
-                Buffer.from(input.scriptSig, 'hex'),
-                input.witness?.map((w) => Buffer.from(w, 'hex')),
-            );
-            if (pubKey) pubKeys.push(pubKey);
-        }
+    // extract the input public keys from the transaction
+    const pubKeys: Buffer[] = [];
+    for (const input of vin) {
+        const pubKey = extractPubKeyFromScript(
+            Buffer.from(input.prevOutScript, 'hex'),
+            Buffer.from(input.scriptSig, 'hex'),
+            input.witness?.map((w) => Buffer.from(w, 'hex')),
+        );
+        if (pubKey) pubKeys.push(pubKey);
+    }
 
-        if (pubKeys.length === 0) return;
+    if (pubKeys.length === 0) return undefined;
 
-        const smallestOutpoint = this.getSmallestOutpoint(vin);
-        const sumOfPublicKeys = Buffer.from(publicKeyCombine(pubKeys, true));
+    const smallestOutpoint = getSmallestOutpoint(vin);
+    const sumOfPublicKeys = Buffer.from(publicKeyCombine(pubKeys, true));
 
-        const inputHash = createTaggedHash(
-            'BIP0352/Inputs',
-            Buffer.concat([smallestOutpoint, sumOfPublicKeys]),
-        );
+    const inputHash = createTaggedHash(
+        'BIP0352/Inputs',
+        Buffer.concat([smallestOutpoint, sumOfPublicKeys]),
+    );
 
-        // A * inputHash
-        const scanTweak = Buffer.from(
-            publicKeyTweakMul(sumOfPublicKeys, inputHash, true),
-        );
+    // A * inputHash
+    const scanTweak = Buffer.from(
+        publicKeyTweakMul(sumOfPublicKeys, inputHash, true),
+    );
 
-        const transaction = new Transaction();
-        transaction.id = txid;
-        transaction.blockHeight = blockHeight;
-        transaction.blockHash = blockHash;
-        transaction.scanTweak = scanTweak.toString('hex');
-        transaction.outputs = eligibleOutputPubKeys;
-        transaction.isSpent = false;
+    return [scanTweak, eligibleOutputPubKeys];
+}
 
-        await this.transactionsService.saveTransaction(transaction);
-    }
+function isP2TR(spk: string): boolean {
+    if (spk.match(/^5120[0-9a-fA-F]{64}$/)) return true;
+}
 
-    private isP2TR(spk: string): boolean {
-        if (spk.match(/^5120[0-9a-fA-F]{64}$/)) return true;
+function getSmallestOutpoint(vins: TransactionInput[]): Buffer {
+    const outpoints = vins.map((vin) => {
+        const n = Buffer.alloc(4);
+        n.writeUInt32LE(vin.vout);
+        return Buffer.concat([Buffer.from(vin.txid, 'hex').reverse(), n]);
+    });
+
+    let smallest = outpoints[0];
+    for (const outpoint of outpoints) {
+        if (outpoint.compare(smallest) < 0) smallest = outpoint;
     }
+    return smallest;
+}
```

Comment:
What do you think about putting this also in `common` folder.

Reply:
i dont think there is a need to, you've already made this exportable.

Reply:
ok, that is fine with me